src/mame/ussr/bk_m.cpp

```diff
@@ -99,10 +204,55 @@ u32 bk_state::screen_update(screen_device &screen, bitmap_ind16 &bitmap, const r
 	{
 		for (u16 x = 0; x < 32; x++)
 		{
-			u16 const code = (y > 63 && mini) ? 0 : m_vram[((y+nOfs) %256)*32 + x];
+			u16 const code = (y > 63 && mini) ? 0 : m_ram[0][((y + nOfs) % 256) * 32 + x];
+			if (m_monitor)
+			{
+				for (u8 b = 0; b < 16; b += 2)
+				{
+					int pixel = (code >> b) & 3;
+					bitmap.pix(y, x * 16 + b) = pixel;
+					bitmap.pix(y, x * 16 + b + 1) = pixel;
+				}
+			}
+			else
+			{
+				for (u8 b = 0; b < 16; b++)
+					bitmap.pix(y, x * 16 + b) = BIT(code, b) << 2;
+			}
+		}
+	}
+	return 0;
+}
+
+TIMER_DEVICE_CALLBACK_MEMBER(bk_state::scanline_callback_bk11)
```

Comment:
Shouldn't that use the standard partial screen update mechanism?

Reply:
On the other hand your screen_update should also clip against the current cliprect otherwise will get over-copying (which translates to poor performance). 

Note that I don't think this is a bad approach of rendering (video chips usually prepare data per scanline, around hblank), but I wouldn't call `TIMER(config, "scantimer")` but rather instantiate an `emu_timer` following the current beam positions.

Reply:
Thanks, I'll take a look for a future PR (have more updates planned for BK hardware).